Mage.Sets/src/mage/cards/c/CurseOfDisturbance.java

```diff
@@ -98,12 +98,12 @@ public boolean checkEventType(GameEvent event, Game game) {
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
-        Permanent enchantment = game.getPermanent(this.getSourceId());
+        Permanent enchantment = game.getPermanentOrLKIBattlefield(this.getSourceId());
         UUID controller = this.getControllerId();
         if (enchantment != null
                 && enchantment.getAttachedTo() != null
                 && game.getCombat().getPlayerDefenders(game).contains(enchantment.getAttachedTo())) {
-            if (game.getCombat().getAttackerId() != controller) {
+            if (!game.getCombat().getAttackerId().equals(controller)) {
                 for (Effect effect: this.getEffects()) {
```

Comment:
You can use  
this.getEffects().setTargetPointer(new FixedTarget(game.getCombat().getAttackerId()));